### ISSUE TITLE
Node name is appended a "1" when reading a trained model

### ISSUE THREAD

**Issue by secretsh** [microsoft/CNTK#3720]
Hi:

  I built a DSSM_LSTM model with BS, there is one node defined like this:
inputAxisK = DynamicAxis()  

  The model runs well during training; but when I run prediction with the trained model, I encountered an error message "Network has no node named 'inputAxisK'."; so I looked back into the log, finding the node name in the model becomes  "inputAxisK1", there is a suffix "1" in the original node name.  
  
   Why could this happen? I pasted the training code as following:

_Train = {
    action = "train"
	outputPath = $modelDir$/model.writeaction1
    outputNodeNames =ce
	
    BrainScriptNetworkBuilder = {
		numNeg              = $NegNumber$
	
	 # Constant tensors
        CONST_GAMMA     = Constant(10)
        CONST_SHIFT     = Constant(1)
        CONST_NEG       = Constant(numNeg)
        
        Click = Constant(1)
		
        inputDim = $vocabSize$
        labelDim = $numLabels$
        embDim = $emb_hid_dim$
        hiddenDim = $emb_hid_dim$
        
        BiRecurrentLSTMLayer {outDim} = {
            F = RecurrentLSTMLayer {outDim, goBackwards=true, init='uniform'}
            G = RecurrentLSTMLayer {outDim, goBackwards=false, init='uniform'}
            apply (x) = Splice(F(x):G(x))
        }.apply
		
		
        
        BLSTMBNLayer {hiddenDim} = {
            F = BiRecurrentLSTMLayer {hiddenDim} 
            G = BatchNormalizationLayer {normalizationTimeConstant=2048} 
            apply (x) = G(F(x)) 
        }.apply
        
        BLSTMBNLayerStack {hiddenDim, numLayers} = 
            if numLayers == 1
            then { 
                apply = BLSTMBNLayer {hiddenDim}
            }.apply
            else { 
                apply(x) = BLSTMBNLayerStack {hiddenDim, numLayers-1}(BLSTMBNLayer {hiddenDim}(x))  
            }.apply
        
        model = Sequential (
            BatchNormalizationLayer {normalizationTimeConstant=2048} :
            BLSTMBNLayerStack {hiddenDim, $numLayers$}
        )
		
		BiRecurrentLSTMLayerK {outDim} = {
            F = RecurrentLSTMLayer {outDim, goBackwards=true, init='uniform'}
            G = RecurrentLSTMLayer {outDim, goBackwards=false, init='uniform'}
            apply (x) = Splice(F(x):G(x))
        }.apply
		
		
        
        BLSTMBNLayerK {hiddenDim} = {
            F = BiRecurrentLSTMLayerK {hiddenDim} 
            G = BatchNormalizationLayer {normalizationTimeConstant=2048} 
            apply (x) = G(F(x)) 
        }.apply
        
        BLSTMBNLayerStackK {hiddenDim, numLayers} = 
            if numLayers == 1
            then { 
                apply = BLSTMBNLayerK {hiddenDim}
            }.apply
            else { 
                apply(x) = BLSTMBNLayerStackK {hiddenDim, numLayers-1}(BLSTMBNLayerK {hiddenDim}(x))  
            }.apply
        
        modelK = Sequential (
            BatchNormalizationLayer {normalizationTimeConstant=2048} :
            BLSTMBNLayerStackK {hiddenDim, $numLayers$}
        )
		
		
        # features
		inputAxisK = DynamicAxis()  #for sequene 
		
        query      = Input {inputDim}
        embed_     = Input {$fixemb_dim$}
        slotLabels = Input {labelDim}
		
		keyword = Input {inputDim,dynamicAxis=inputAxisK}
		embed_k = Input {$fixemb_dim$,dynamicAxis=inputAxisK}

        # model application
        z_q = LinearLayer {embDim, init='uniform', bias=false} (query) 
        z_in = Splice(z_q:embed_)
        z_e = model (z_in)
		z_e_last = BS.Sequences.Last(z_e)
		
		z_k = LinearLayer {embDim, init='uniform', bias=false} (keyword) 
		z_k_in = Splice (z_k:embed_k)
		z_k_LSTM = modelK (z_k_in)
		z_k_last = BS.Sequences.Last(z_k_LSTM)
		z_e_last1 = ReconcileDynamicAxis (z_e_last, z_k_last)

		labels = BS.Sequences.Last (slotLabels)
		
		c   = CosDistanceWithNegativeSamples(z_e_last1, z_k_last, CONST_SHIFT, CONST_NEG)
		dc = Scale(CONST_GAMMA, c)
		
        errs   = Slice(0, 1, c, axis=1, tag="output")
        ce  = CrossEntropyWithSoftmax(labels , dc , tag="criterion")
		
        criterionNodes = (ce)
		#featureNodes    = (query)
        labelNodes      = (slotLabels)
        evaluationNodes = (errs)
        outputNodes     = (errs)
    }

    SGD = {
        maxEpochs = $epochs$ ; epochSize = 10000000

        minibatchSize = 64

        learningRatesPerSample = $learningRates$ 
        gradUpdateType = $grad_type$
        fsAdagradTargetAvDenom = 0.0025
        gradientClippingWithTruncation = true ; clippingThresholdPerSample = 15.0

        firstMBsToShowResult = 1 ; numMBsToShowResult = 100
    }

    reader = {
        #readerType = "CNTKTextFormatReader"
        #file = "$train_file$"
		 
		verbosity = 0
        randomize = true
		deserializers = (

		  {type = "CNTKTextFormatDeserializer" ; module = "CNTKTextFormatReader"
		  file = "$train_file$"
		  miniBatchMode = "partial"
		  input = {
				query        = { alias = "F" ; dim = $vocabSize$ ;  format = "sparse" }
				embed_        = { alias = "Se" ; dim = $fixemb_dim$ ;  format = "sparse" }
				slotLabels   = { alias = "L" ; dim = $numLabels$ ;  format = "sparse" }
				#keyword        = { alias = "F2" ; dim = $vocabSize$ ;  format = "sparse" }
				#embed_k        = { alias = "S2" ; dim = $fixemb_dim$ ;  format = "sparse" }
			}} :
			
			{type = "CNTKTextFormatDeserializer" ; module = "CNTKTextFormatReader"
		  file = "$train_file1$"
		  miniBatchMode = "partial"
		  input = {
				#query        = { alias = "F" ; dim = $vocabSize$ ;  format = "sparse" }
				#embed_        = { alias = "Se" ; dim = $fixemb_dim$ ;  format = "sparse" }
				#slotLabels   = { alias = "L" ; dim = $numLabels$ ;  format = "sparse" }
				keyword        = { alias = "F2" ; dim = $vocabSize$ ;  format = "sparse" }
				embed_k        = { alias = "S2" ; dim = $fixemb_dim$ ;  format = "sparse" }
			}}

		)
    }

}_
